### PR TITLE
Remove Roberta Dependencies from XLM Roberta Flax and Tensorflow models

### PR DESCRIPTION
**Pull request by samuelzxu**
# What does this PR do?

This removes Roberta dependencies from XLM Roberta Flax and Tensorflow Models.

I'm a bit confused about whether the `name` parameter to `TFXLMRobertaMainLayer` should be `xml-roberta` or `xml_roberta` - I've gome with `xml-roberta` for now.
i.e. `self.XLMRoberta = TFXLMRobertaMainLayer(config, name="xlm-roberta")`

Fixes #19303

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@sgugger

### CODE REVIEW

File: src/transformers/models/xlm_roberta/modeling_tf_xlm_roberta.py [hunk 1]
```diff
@@ -78,104 +119,1654 @@
...
+
+    # Copied from transformers.models.bert.modeling_tf_bert.TFBertForTokenClassification.serving_output
+    def serving_output(self, output: TFTokenClassifierOutput) -> TFTokenClassifierOutput:
+        hs = tf.convert_to_tensor(output.hidden_states) if self.config.output_hidden_states else None
+        attns = tf.convert_to_tensor(output.attentions) if self.config.output_attentions else None
+
+        return TFTokenClassifierOutput(logits=output.logits, hidden_states=hs, attentions=attns)
+
+
+# Copied from transformers.models.roberta.modeling_tf_roberta.TFRobertaForQuestionAnswering with Roberta->XLMRoberta
```
Comment (sgugger) [inline src/transformers/models/xlm_roberta/modeling_tf_xlm_roberta.py:1672]: This needs to be just before the class decoration.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this! The test that checks the copies is failing because some of the `# Copied from` statement are put before the decorators instead of between the decorator and the class/function def.

For instance:
```py
# Copied from xxx
@add_start_docstring(yyy)
class Model
```
doesn't work, it needs to be
```py
@add_start_docstring(yyy)
# Copied from xxx
class Model
```

**Comment by sgugger** [comment]
Thanks! There are a couple of places where the copy does not match the original. You can test locally with `make repo-consistency` to get the failing tests. Let me know if you need help!

**Comment by SamuelzXu** [comment]
HI @sgugger, thanks for the advice! I've tried `make repo-consistency` but it looks like the version of jax that was installed during `pip -e ".[dev]"` causes `RuntimeError: jaxlib is version 0.1.75, but this version of jax requires version >= 0.3.0.`

**Comment by SamuelzXu** [comment]
Ah, it's a problem with my original dev install. I'll reinstall and see how it goes.

**Comment by SamuelzXu** [comment]
I can't figure out this last copy error, could you help me out? Thanks.

**Comment by SamuelzXu** [comment]
Got it, thanks for the tip

**Comment by SamuelzXu** [comment]
@sgugger I'm stuck on this error message as a part of `make repo-consistency`: 
```
python utils/check_inits.py
Traceback (most recent call last):
  File "/home/ziggy/dev/transformers/utils/check_inits.py", line 299, in <module>
    check_all_inits()
  File "/home/ziggy/dev/transformers/utils/check_inits.py", line 238, in check_all_inits
    raise ValueError("\n\n".join(failures))
ValueError: Problem in src/transformers/__init__.py, both halves do not define the same objects.
Differences for tf backend:
  TFXLMRobertaForCausalLM in _import_structure but not in TYPE_HINT.
  TFXLMRobertaPreTrainedModel in _import_structure but not in TYPE_HINT.
Differences for flax backend:
  FlaxXLMRobertaForCausalLM in _import_structure but not in TYPE_HINT.
  FlaxXLMRobertaPreTrainedModel in _import_structure but not in TYPE_HINT.

Problem in src/transformers/models/xlm_roberta/__init__.py, both halves do not define the same objects.
Differences for flax backend:
  FLAX_XLM_ROBERTA_PRETRAINED_MODEL_ARCHIVE_LIST in TYPE_HINT but not in _import_structure.
make: *** [Makefile:41: repo-consistency] Error 1
```

I don't know where the variable `TYPE_HINT` is, it doesn't seem to be anywhere in the entire repo apart from this error message.

File: src/transformers/__init__.py [hunk 2]
```diff
@@ -3133,12 +3133,14 @@
     _import_structure["models.xlm_roberta"].extend(
         [
             "TF_XLM_ROBERTA_PRETRAINED_MODEL_ARCHIVE_LIST",
+            "TFXLMRobertaForCausalLM",
```
Comment (sgugger) [inline src/transformers/__init__.py:3136]: You need to add those to the second half of the init as well (beneath `if TYPE_CHECKING`.

File: src/transformers/__init__.py [hunk 3]
```diff
@@ -3421,6 +3423,8 @@
             "FlaxXLMRobertaForSequenceClassification",
             "FlaxXLMRobertaForTokenClassification",
             "FlaxXLMRobertaModel",
+            "FlaxXLMRobertaForCausalLM",
```
Comment (sgugger) [inline src/transformers/__init__.py:3426]: Same for those.

**Review by sgugger — COMMENTED** [review]
This is because you only defined the objects in half the init. Left comments to help :-)

**Comment by SamuelzXu** [comment]
Ah never mind, I found them. Thanks!

**Comment by SamuelzXu** [comment]
I'm confused by the error that's showing now - the doc builder can't find `TFXLMRobertaForCausalLM.forward` , which I don't think exists because it's tensorflow...

File: docs/source/en/model_doc/xlm-roberta.mdx [hunk 4]
```diff
@@ -146,6 +146,11 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 [[autodoc]] TFXLMRobertaModel
     - call
 
+## TFXLMRobertaForCausalLM
+
+[[autodoc]] TFXLMRobertaForCausalLM
+    - forward
```
Comment (sgugger) [inline docs/source/en/model_doc/xlm-roberta.mdx:152]:
```suggestion
    - call
```

File: docs/source/en/model_doc/xlm-roberta.mdx [hunk 5]
```diff
@@ -176,6 +181,11 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 [[autodoc]] FlaxXLMRobertaModel
     - __call__
 
+## FlaxXLMRobertaForCausalLM
+
+[[autodoc]] FlaxXLMRobertaForCausalLM
+    - forward
```
Comment (sgugger) [inline docs/source/en/model_doc/xlm-roberta.mdx:187]:
```suggestion
    - __call__
```

**Review by sgugger — COMMENTED** [review]
Be sure to use the proper method names in the doc file :-)

**Comment by SamuelzXu** [comment]
Finally, no errors 😆

**Comment by SamuelzXu** [comment]
Thanks so much for all the help!

File: utils/check_repo.py [hunk 6]
```diff
@@ -206,6 +206,8 @@
     "FlaxCLIPTextModel",
     "FlaxCLIPVisionModel",
     "FlaxWav2Vec2ForCTC",
+    "FlaxXLMRobertaForCausalLM",
+    "TFXLMRobertaForCausalLM",
```
Comment (sgugger) [inline utils/check_repo.py:210]: Oh those should be put in the auto-mappings. You just have to set them in `models.auto.modeling_flax_auto` and `models.auto.modeling_tf_auto`.

**Review by sgugger — COMMENTED** [review]
Thanks for all your work! Last step, let's just add the two new models in the proper auto mapping (instead of putting them in the list of exceptions in check_repo) and we should be good to go!

**Review by sgugger — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**